src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4104,10 +4104,11 @@ internal PSModuleInfo LoadRequiredModule(PSModuleInfo currentModule, ModuleSpeci
                     if (currentModule != null)
                     {
                         requiredModules.Add(new ModuleSpecification(currentModule), new List<ModuleSpecification> { requiredModuleSpecification });
+                        requiredModules.Add(requiredModuleSpecification, new List<ModuleSpecification> ( requiredModuleInfo.RequiredModulesSpecification ));
```

Comment:
Originally I was thinking that both addition should be bundled together; but thinking more about it - it is better to move the new addition to its own 'if' check. Updated.